packages/infield-button/src/spectrum-infield-button.css

```diff
@@ -11,144 +11,190 @@ governing permissions and limitations under the License.
+    --spectrum-infield-button-stacked-fill-padding-outer: var(--spectrum-in-field-button-outer-edge-to-disclosure-icon-stacked-medium);
+    --spectrum-infield-button-stacked-fill-padding-inner: var(--spectrum-in-field-button-inner-edge-to-disclosure-icon-stacked-medium);
+    --spectrum-infield-button-icon-color: var(--spectrum-neutral-content-color-default);
+    --spectrum-infield-button-icon-color-hover: var(--spectrum-neutral-content-color-hover);
+    --spectrum-infield-button-icon-color-down: var(--spectrum-neutral-content-color-down);
+    --spectrum-infield-button-icon-color-key-focus: var(--spectrum-neutral-content-color-key-focus);
+    --spectrum-infield-button-fill-justify-content: center;
 }
 
-.fill {
-    block-size: 100%;
-    inline-size: 100%;
-    background-color: var(--mod-infield-button-background-color, var(--spectrum-infield-button-background-color));
-    border-width: var(--mod-infield-button-border-width, var(--spectrum-infield-button-border-width));
-    border-style: solid;
-    border-color: var(--highcontrast-infield-button-border-color, var(--mod-infield-button-border-color, var(--spectrum-infield-button-border-color)));
-    padding: var(--mod-infield-button-fill-padding, var(--spectrum-infield-button-fill-padding));
-    border-start-start-radius: var(--mod-infield-button-border-radius, var(--spectrum-infield-button-border-radius));
-    border-start-end-radius: var(--mod-infield-button-border-radius, var(--spectrum-infield-button-border-radius));
-    border-end-end-radius: var(--mod-infield-button-border-radius, var(--spectrum-infield-button-border-radius));
-    border-end-start-radius: var(--mod-infield-button-border-radius, var(--spectrum-infield-button-border-radius));
+:host([disabled]) {
+    --mod-infield-button-background-color: var(--mod-infield-button-background-color-disabled, var(--spectrum-disabled-background-color));
+    --mod-infield-button-background-color-hover: var(--mod-infield-button-background-color-hover-disabled, var(--spectrum-disabled-background-color));
+    --mod-infield-button-background-color-down: var(--mod-infield-button-background-color-down-disabled, var(--spectrum-disabled-background-color));
+    --mod-infield-button-border-color: var(--mod-infield-button-border-color-disabled, var(--spectrum-infield-button-border-color));
+    --mod-infield-button-icon-color: var(--mod-infield-button-icon-color-disabled, var(--spectrum-disabled-content-color));
+    --mod-infield-button-icon-color-hover: var(--mod-infield-button-icon-color-hover-disabled, var(--spectrum-disabled-content-color));
+    --mod-infield-button-icon-color-down: var(--mod-infield-button-icon-color-down-disabled, var(--spectrum-disabled-content-color));
+    --mod-infield-button-icon-color-key-focus: var(--mod-infield-button-icon-color-key-focus-disabled, var(--spectrum-disabled-content-color));
```

Comment:
These should be reverted back also!

Reply:
If a mod is pointing to another modifier in the stack, it's not an issue, it's an intentional pattern that subs in the more specific modifier so that customers don't end up with unexpected rendering results. Imagine you are using both --mod-*-background-color and --mod-*-background-color-disabled but the disabled mod isn't picking up because we haven't updated the var stack?